tools/README.md

```diff
@@ -1,21 +1,133 @@
+
+```
+pip install -r requirements.txt
+```
+
+Also, to create the directory authority's certificate, this script rely
+on the program `tor-gencert` that you will have to install on your
+machine. For a debian based distribution, this program is usually
+installed by the package `tor`:
+
+```
+sudo apt-get install tor
+```
+
+## Generating Directory Info
+
+We provide a Makefile allowing you to generate all the files you need
+for running Arti-rest, but you still need to setup a password to encrypt
+some private data of the custom authority via an environment variable:
+
+```
+export DIR_AUTH_PASSWORD='dummypassword'
+make
+```
+
+We configured it to place the files required by Arti-rest in the
+sub-directory `directory-cache`, and the private data of the custom
+directory authority in the sub-directory `authority-private`.
+
+Once the data generated, you can use the subdirectory `directory-cache`
+as the `cache_dir` argument.
+
+### Custom Directory Authority
+
+As the directory information needs to be signed, you will need to
+create a custom directory authority with a valid certificate. The
+script can generate a custom directory authority with a valid
+certificate by using its `generate-certificate` sub-command.
+
+This step is required, but you only need to do it to create the
+custom directory authority and when renewing its certificate.
+
+As `tor-gencert` requires the identity key of the directory
+authority to be encrypted, you need to pass a password to the script
+via an environment variable.
+
+```
+export DIR_AUTH_PASSWORD='dummypassword'
+```
+
+Then you can call the script.
+
+```
+python3 gen_fresh_dirinfo.py generate-certificate \
+  --authority-identity-key authority_identity_key \
+  --authority-signing-key authority_signing_key \
+  --authority-certificate certificate.txt \
+  --authority-v3ident authority.txt \
+  --authority-name spring \
+  --certificate-lifetime 12
+```
+
+Which creates 4 files:
+
+- The identity key of the authority which you should keep private in a
+  secure location.
+- The signing key of the authority which you should keep private and
+  that you will need for generating the directory information.
+- The certificate of the authority which you will need for generating
+  the directory information and which you will need to provide to the
+  Arti-rest library.
+- A small file containing the v3ident identifier of the authority which
+  you will need to provide to the Arti-rest library.
+
+
+### Generate Directory Information
+
+To generate the directory information, you will need to run the
+`generate-dirinfo` sub-command this script regularly on a trusted
+server.
+
+```
+python3 gen_fresh_dirinfo.py generate-dirinfo \
+  --authority-signing-key authority_signing_key \
+  --authority-name spring \
+  --authority-contact 'SPRING Lab at EPFL' \
+  --authority-certificate certificate.txt \
+  --consensus consensus.txt \
+  --consensus-lifetime 7 \
```

Comment:
Same question as for `certificate-lifetime` - here I think the consensus-lifetime` is in days, and I hope the certificate is valid longer than that...